identity/service_principals.go

```diff
@@ -0,0 +1,300 @@
+package identity
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/databrickslabs/databricks-terraform/common"
+)
+
+// NewServicePrincipalsAPI creates ServicePrincipalsAPI instance from provider meta
+func NewServicePrincipalsAPI(m interface{}) ServicePrincipalsAPI {
+	return ServicePrincipalsAPI{C: m.(*common.DatabricksClient)}
+}
+
+// ServicePrincipalsAPI exposes the scim servicePrincipal API
+type ServicePrincipalsAPI struct {
+	C *common.DatabricksClient
+}
+
+// ServicePrincipalEntity entity from which resource schema is made
+type ServicePrincipalEntity struct {
+	ApplicationId           string `json:"application_id"`
+	DisplayName             string `json:"display_name,omitempty" tf:"computed"`
+	Active                  bool   `json:"active,omitempty"`
+	AllowClusterCreate      bool   `json:"allow_cluster_create,omitempty"`
+	AllowInstancePoolCreate bool   `json:"allow_instance_pool_create,omitempty"`
+}
+
+func (u ServicePrincipalEntity) toRequest() ScimServicePrincipal {
```

Comment:
instead of `u` (first letter for user), do `sp`

Reply:
ad0a9e4065935cd6cda6064ce5ad45ca445f9bac